extras/test_hull_performace.cpp

```diff
@@ -0,0 +1,175 @@
+    vertices[i] = cgalMesh.add_vertex(Point(vert.x, vert.y, vert.z));
+  }
+
+  for (auto &triVert : maniMesh.triVerts) {
+    std::vector<Vertex> polygon{vertices[triVert[0]], vertices[triVert[1]],
+                                vertices[triVert[2]]};
+    cgalMesh.add_face(polygon);
+  }
+}
+
+// Prints the volume and area of the manifold and the convex hull of our
+// implementation,
+void PrintVolArea(Manifold &manfiold_obj, Manifold &hullManifold) {
+  std::cout << manfiold_obj.GetProperties().volume << ",";
+  std::cout << hullManifold.GetProperties().volume << ",";
+  std::cout << manfiold_obj.GetProperties().surfaceArea << ",";
+  std::cout << hullManifold.GetProperties().surfaceArea << ",";
+  std::cout << manfiold_obj.NumTri() << ",";
+  std::cout << hullManifold.NumTri() << ",";
+  return;
+}
+
+// Prints the volume and area of the manifold and Convex Hull of CGAL
+// Implementation
+void PrintVolAreaCGAL(TriangleMesh &cgalMesh, TriangleMesh &cgalHull) {
+  std::cout << CGAL::Polygon_mesh_processing::volume(cgalMesh) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::volume(cgalHull) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::area(cgalMesh) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::area(cgalHull) << ",";
+  std::cout << std::distance(cgalMesh.faces_begin(), cgalMesh.faces_end())
+            << ",";
+  std::cout << std::distance(cgalHull.faces_begin(), cgalHull.faces_end())
+            << ",";
+  return;
+}
+
+void PrintManifold(Manifold input, Manifold (Manifold::*hull_func)() const) {
+  auto start = std::chrono::high_resolution_clock::now();
+  Manifold hullManifold = (input.*hull_func)();
+#ifdef MANIFOLD_EXPORT
+  if (options.exportModels) {
+    ExportMesh("testHullManifold.glb", hullManifold.GetMesh(), {});
+  }
+#endif
+  auto end = std::chrono::high_resolution_clock::now();
+  PrintVolArea(input, hullManifold);
+  std::chrono::duration<double> elapsed = end - start;
+  std::cout << elapsed.count() << " sec";
+}
+
+void PrintCGAL(Manifold input, Manifold (Manifold::*hull_func)() const) {
+  TriangleMesh cgalInput;
+  manifoldToCGALSurfaceMesh(input, cgalInput);
+  std::vector<Point> points;
+  for (const auto &vert : cgalInput.vertices()) {
+    points.push_back(cgalInput.point(vert));
+  }
+  // Convex Hull
+  TriangleMesh cgalHull;
+  auto start = std::chrono::high_resolution_clock::now();
+  CGAL::convex_hull_3(points.begin(), points.end(), cgalHull);
+  auto end = std::chrono::high_resolution_clock::now();
+  PrintVolAreaCGAL(cgalInput, cgalHull);
+  std::chrono::duration<double> elapsed = end - start;
+  std::cout << elapsed.count() << " sec";
+}
+
+// Constructs a Menger Sponge, and tests the convex hull implementation on it
+// (you can pass the specific hull implementation to be tested). Comparing the
+// volume and surface area with CGAL implementation, for various values of
+// rotation
+void MengerTestHull(Manifold (Manifold::*hull_func)() const, float rx, float ry,
```

Comment:
This is fine for comparison to CGAL, but I'm actually more interested in adding to our own testing framework. Can you build tests like this directly into `hull_test.cpp` for just Manifold's implementation?

Reply:
So we just want to keep this testing script for the tests on datasets, and these single tests, I can shift to `hull_test.cpp`?